presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java

```diff
@@ -190,6 +190,12 @@ public void testDuplicatedRowCreateTable()
                 "line 1:49: Column name 'orderkey' specified more than once");
     }
 
+    @Test
+    public void testVarcharIsDistinctFrom()
```

Comment:
This should go in AbstractTestQueries, not in the connector integration tests.

Reply:
I was not able to simply create a non-varchar(n) table on AbstractTestQueries. `CAST` also didn't help to reproduce the issue as `DomainTranslator.visitComparisonExpression` has the own coercing. Let me try again.

Reply:
I could use tables under information_schema